Use container queries for unified layout

avatus opened the pull request:
This fixes the layout to better match the breakpoints. We were doing a bit of guesswork before with the sidebar (the bane of our layout) and so I opted in to using [container queries](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_container_queries)

Using a container query makes our development experience match what we are "thinking about" when we talk about breakpoints in our content containers. the downside is that `styled-components` v5 doesn't support it. it's in v6 but that introduced a bunch of breaking changes to our codebase so rather than doing ALL OF THAT for this one page, i just opted to importing some basic css.

When the sidebar goes away we can move back to media queries very easily. 


https://github.com/gravitational/teleport/assets/5201977/5d92da5e-7bc1-454b-a778-73481431ccef

avatus committed:
Use container queries for unified layout

kimlisa reviewed (commented):
unfortunately, we can't do this, unless we can scope it to just unified resources (even then i haven't tested out the min sizes of all available states in unified resource)

it's because our pages are optimized with the `min-width` we set it. allowing the pages to shrink all the way will break them, lose readability, and we lose horizontal scrolling. 

the way we have it currently (having a fixed min-width) allows you to still browse around in smaller screen, you just have to scroll around.

we should punt this for later, when we design for responsiveness (not every page is suited for it) and when we do design for it, it should be like a page by page change 

i think among the first change should be to hide the sidebar b/c it takes up so much space and i think there are talks to minify it

avatus commented:
I'll see if I can scope it to unified resources exclusively before I punt it

avatus commented:
@kimlisa ok, i pushed an experimental commit https://github.com/gravitational/teleport/pull/31518/commits/8ec02bdb0206b6b0a6928ab3f3456f1f5b525831

What this does is adds back in that `ContentMinWidth` class but instead, wrap the `Route` with it. Then I added an optional prop to `useOwnLayout` which would render that route without the contentminwidth. its a little escape hatch that we can optionally use on a per route basis (and maybe even use going forward for new, responsive pages). 

I did light testing just to see if the idea worked. I'm sure more testing will need to be done but I didn't wanna deep dive until I got your thoughts.

ryanclark commented:
We should really fix our main layout. What we currently have isn't ideal:

<img width="722" alt="image" src="https://github.com/gravitational/teleport/assets/7922109/04bed701-a9c9-407e-96c9-d2948e4ba5ff">

The page doesn't scroll, the inner divs do. This is fine for the sidebar but for the content, it should be the page that scrolls. I wanted to take some time to fix this and implement everything properly, but I never got round to it.

That being said, I'm fine with this approach for this page until we can fix it properly

avatus commented:
I'm hoping that when we get to removing that sidebar, we can fix up all the layout issues (and remove this "fix") but that could be awhile 🥴 
Thanks for the input, Ryan!

rudream commented on web/packages/teleport/src/types.ts:
```diff
@@ -87,6 +87,9 @@ export interface TeleportFeatureRoute {
   component: React.FunctionComponent;
+  // useOwnLayout is used to ignore the ContentMinWidth that wraps
+  // every component. default = false
+  useOwnLayout?: boolean;
```
nit: maybe rename this to `hasOwnLayout` or something since `useXXX` naming typically refers to hooks

avatus replied:
good call

kimlisa reviewed (commented):
this is still too broken for my liking. i _think_ there are some people who actually tries to access an app, through phone: https://github.com/gravitational/teleport/issues/20996

currently, at it's most minimal size, it's not very friendly to search for apps 

<img width="488" alt="image" src="https://github.com/gravitational/teleport/assets/43280172/25f00137-db90-4441-8938-5ce42cc14eaa">

also, other pages there are min-widths, but the top bar shrinks, so going to the far left of the screen, the user drop down menu is not where i expect it to be.

If we really wanted this to work, we need to set a min-width to unified resources screen, and not let the top bar be affected for others

avatus commented:
Ok, updated. Removed any min-width stuff. Sorry for the rebase, had a nasty conflict but the new pr is really small now! Basically i'm just adding the container query to unified resource page and thats it lol. Let me know if it's any better. dismissing old reviews
